jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -458,7 +446,7 @@ void initialize_nondet_string_fields(
   code.add(code_assumet(binary_relation_exprt(length_expr, ID_ge, min_length)));
 
   // assume (length_expr <= max_input_length)
-  if(max_nondet_string_length <= max_value(length_expr.type()))
+  if(max_nondet_string_length <= bv_spect(length_expr.type()).max_value())
```

Comment:
How about `to_bitvector_type(length_expr.type()).largest()`?

Reply:
Good point, why does the class `bv_spect` even (publicly) exist if this functionality is already made available by the typet?
I changed it to your suggestion for now. Depending on what you think about my previous question I might change `to_bitvector_type` to something more specific.
I also found another place where `bv_spect` was used to get the max value, and replaced this with `bitvector_type::largest` as well.

Reply:
Following @kroening 's comment above, this is now `to_integer_bitvector_type(...)` (and similarly for your other comment).